chef/cookbooks/provisioner/recipes/shell-vars.rb

```diff
@@ -0,0 +1,41 @@
+# Copyright 2015, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+env = node[:provisioner][:config][:environment]
+
+node_vars = []
+role_vars = []
+
+search(:node, "provisioner_config_environment:#{env}") do |n|
+  aliaz = n["crowbar"]["display"]["alias"] rescue nil
+  next unless aliaz && !aliaz.empty?
+  short = aliaz.gsub("node", "n")
+  node_vars.push [short, n.fqdn]
+
+  role = "crowbar-" + n.fqdn.gsub(".", "_")
+  role_vars.push [short + "r", role]
```

Comment:
Sorry, I disagree.  Even SEs would find this useful.  And it doesn't do any harm to have it everywhere (see updated version), so why not keep it?  If anything, I'd prefer to drop the whole file from all non-admin nodes.